api/lib/opentelemetry/instrumentation/base_instrumentation.rb

```diff
@@ -39,27 +39,27 @@ module Instrumentation
     #   end
     # end
     #
-    # The adapter name and version will be inferred from the namespace of the
-    # class. In this example, they'd be 'OpenTelemetry::Adapters::Sinatra' and
-    # OpenTelemetry::Adapters::Sinatra::VERSION, but can be explicitly set using
-    # the +adapter_name+ and +adapter_version+ methods if necessary.
+    # The instrumentation name and version will be inferred from the namespace of the
+    # class. In this example, they'd be 'OpenTelemetry::Instrumentation::Sinatra' and
+    # OpenTelemetry::Instrumentation::Sinatra::VERSION, but can be explicitly set using
+    # the +instrumentation_name+ and +instrumetation_version+ methods if necessary.
     #
-    # All subclasses of OpenTelemetry::Instrumentation::Adapter are automatically
+    # All subclasses of OpenTelemetry::Instrumentation::BaseInstrumentation are automatically
     # registered with OpenTelemetry.instrumentation_registry which is used by
     # SDKs for instrumentation discovery and installation.
     #
-    # Instrumentation libraries can use the adapter subclass to easily gain
+    # Instrumentation libraries can use the instrumentation subclass to easily gain
     # a reference to its named tracer. For example:
     #
-    # OpenTelemetry::Adapters::Sinatra.instance.tracer
+    # OpenTelemetry::Instrumentation::Sinatra.instance.tracer
     #
-    # The adapter class establishes a convention for disabling an adapter
-    # by environment variable and local configuration. An adapter disabled
+    # The instrumention class establishes a convention for disabling an instrumentation
+    # by environment variable and local configuration. An instrumentation disabled
     # by environment variable will take precedence over local config. The
     # convention for environment variable name is the library name, upcased with
     # '::' replaced by underscores, and '_ENABLED' appended. For example:
-    # OPENTELEMETRY_ADAPTERS_SINATRA_ENABLED = false.
-    class Adapter
+    # OPENTELEMETRY_INSTRUMENTATIONS_SINATRA_ENABLED = false.
+    class BaseInstrumentation
```

Comment:
I wouldn't be opposed to calling this just `Base`, but `BaseInstrumentation` is also fine. I wouldn't change it unless others chime in with the same preference.

Reply:
I think  `BaseInstrumentation` is a bit more descriptive than just `Base` and matches the python repo's approach but, i too do not have strong feelings here

Reply:
We settled on `Base` on the SIG call. (Recording for posterity)